sematic/plugins/abstract_storage.py

```diff
@@ -0,0 +1,50 @@
+# Standard Library
+import abc
+import enum
+
+
+class StorageMode(enum.Enum):
+    READ = "read"
+    WRITE = "write"
+
+
+class AbstractStorage(abc.ABC):
+    """
+    Abstract base class to represent a key/value storage engine.
+    """
```

Comment:
If we want to support storing logs somewhere other than S3 (e.g. GCS), we will need this abstraction to also include `set_from_file`, `get_line_stream`, and a couple other methods.

Reply:
Yes, for now the logs code will use `S3Storage` directly, as it does currently. If we add these interfaces here, it means we need to implement them on all plug-ins which is out of scope here.

Reply:
Fair enough